### PR TITLE
Accept color name "magenta" instead of "purple"

### LINKED ISSUE #11456: "Magenta" is called "Purple" in WT
### Windows Terminal version (or Windows build number)

All versions

### Other Software

_No response_

### Steps to reproduce

Terminal calls "Magenta" by the wrong name, instead it is called "Purple".

### Expected Behavior

The expected behavior is calling it "Magenta" instead of "Purple".

### Actual Behavior

Terminal calls the color in question "Purple", instead of "Magenta".

### PR DESCRIPTION
**Pull request by matthewd673**
When loading color schemes from Json, check if GetValueForKey failed. If
it did, and we were searching for the colors "purple"/"brightPurple",
swap out the color name with "magenta"/"brightMagenta" and try again.
This was tested manually by creating a new color scheme using the colors
"magenta"/"brightMagenta" and loading it, then printing colored text in
the terminal to assure that the colors were correctly assigned as
purple.

This changes the color loader to use an index pair table to add support
for alternate color names.

## Validation Steps Performed
- Manually edit settings.json, creating a new color scheme with colors
  "magenta" and "brightMagenta"
- View the color scheme in settings - the colors will appear as "purple"
  and "brightPurple" in the UI
- Run a program that prints colored text, purple and brightPurple text
  will appear in the colors defined as magenta and brightMagenta
- Modify the color scheme in the settings UI and save it, the colors
  will be saved as "purple" and "brightPurple" *(I think this is the
  right behavior, since calling them "magenta" is technically a
  mistake)*
- Repeat the steps above with a normal color scheme - colors named
  "purple" and "brightPurple" still load properly

Closes #11456

### CODE REVIEW

**Comment by lhecker** [comment]
If most people edit their schemes in the settings UI, is there a benefit of changing the label in the settings.json only, without also changing it in the settings UI? How did the "magenta" get into the settings.json in the first place?

In other words, if #11456 is about "purple" vs. "magenta" in general (because ECMA-48 calls it "magenta"), should we prefer changing the naming throughout the entire application to be consistent? Otherwise I'm worried whether only changing the settings.json parser will have any positive effect on the issue.

File: src/cascadia/TerminalSettingsModel/ColorScheme.cpp [hunk 1]
```diff
@@ -100,7 +100,25 @@ bool ColorScheme::_layerJson(const Json::Value& json)
...
+            }
+            else if (tableColorName == "brightPurple")
+            {
+                tableColorName = "brightMagenta";
+            }
+
+            valueForKeyIsValid = JsonUtils::GetValueForKey(json, tableColorName, til::at(_table, i));
+        }
+
+        isValid &= valueForKeyIsValid;
```
Comment (lhecker) [inline src/cascadia/TerminalSettingsModel/ColorScheme.cpp:121]: BTW the second lookup is only necessary if the `tableColorName` value changed.

It's technically possible to solve this issue without additional `if` conditions, by extending our code in its declarative style. Mapping arrays are pretty cool after all! 😄

```cpp
static constexpr std::array<std::pair<std::string_view, size_t>, 18> ColorTableMapping{{
    // Main mappings which are the ones written to settings.json
    {"black", 0},
    {"red", 1},
    {"green", 2},
    {"yellow", 3},
    {"blue", 4},
    {"purple", 5},
    {"cyan", 6},
    {"white", 7},
    {"brightBlack", 8},
    {"brightRed", 9},
    {"brightGreen", 10},
    {"brightYellow", 11},
    {"brightBlue", 12},
    {"brightPurple", 13},
    {"brightCyan", 14},
    {"brightWhite", 15},

    // Extra fallback mappings during reading due to GH#11456
    {"magenta", 5},
    {"brightMagenta", 13},
}};

bool ColorScheme::_layerJson(const Json::Value& json)
{
    // ...

    // Required fields
    size_t colorCount = 0;
    for (const auto& [key, index] : ColorTableMapping)
    {
        colorCount += JsonUtils::GetValueForKey(json, key, til::at(_table, index));
    }
    // We're still valid if we read exactly 16 colors.
    isValid &= colorCount == 16;

    return isValid;
}

Json::Value ColorScheme::ToJson() const
{
    // ...

    for (size_t i = 0; i < 16; ++i)
    {
        const auto& key = til::at(ColorTableMapping, i).first;
        JsonUtils::SetValueForKey(json, key, til::at(_table, i));
    }

    return json;
}
```

Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/ColorScheme.cpp:121]: (this is elegant, I love it)

**Comment by zadjii-msft** [comment]
> How did the "magenta" get into the settings.json in the first place?

There are other tools that generate terminal color schemes for terminal emulators, and we're a little different than the others in regard to that key. The original report of this (https://github.com/microsoft/terminal/issues/2641#issuecomment-528348251) greatly predates the SUI, when this would have been more of an issue. 

I don't think it's necessarily super valuable to change the whole app over to Magenta instead of purple (that would of course cause a rewrite of everyone's settings). I do think it's valuable to gracefully accept `magenta` though.

**Review by lhecker — APPROVED** [review]

**Comment by matthewd673** [comment]
Thank you for the feedback.
In terms of the usefulness of this fix, besides the original complaint https://github.com/microsoft/terminal/issues/11456#issuecomment-938165610, in the case that "purple" is renamed to "magenta" across the entire program, this code could support legacy configurations that still say "purple".

**Review by zadjii-msft — APPROVED** [review]
❤️

**Pull request merged**